videostore/videostore.go

```diff
@@ -119,9 +133,54 @@ func (s *service) Close(_ context.Context) error {
 		return err
 	}
 	s.vs.Close()
+	s.workers.Stop()
 	return nil
 }
 
+// GetVideo streams video chunks between the given timestamps.
+func (s *service) GetVideo(
+	ctx context.Context,
+	startTime, endTime time.Time,
+	videoCodec, videoContainer string,
+	_ map[string]interface{},
+) (chan *video.Chunk, error) {
+	s.logger.Debugf(
+		"GetVideo called with startTime: %s, endTime: %s, videoCodec: %s, videoContainer: %s",
+		startTime.Format(time.RFC3339),
+		endTime.Format(time.RFC3339),
+		videoCodec,
+		videoContainer,
+	)
+
+	req := &videostore.FetchRequest{
+		From: startTime,
+		To:   endTime,
+	}
+	ch := make(chan *video.Chunk)
+
+	s.workers.Add(func(workerCtx context.Context) {
```

Comment:
is using workers necessary here? What are the advantages to using them instead of `ManagedGo`? I don't see why we would unless we have a pool of workers that makes it more efficient.

Reply:
Ok, moved to using `context.AfterFunc` to bind the request ctx and the workerCtx together. I do not like these complicated context handlers, but I think it is necessary here.

Reply:
@oliviamiller Unfortunately, I ended up moving back to the merged Ctx model

Realized my concurrent GetVideo test was failing. See [here](https://github.com/viam-modules/viamrtsp/actions/runs/19970010084/job/57272060433?pr=164)


If the emit never fires, FetchStream never gets notified of the context shutdown.